src/libYARP_dev/include/yarp/dev/IJoypadController.h

```diff
@@ -253,8 +253,10 @@ class YARP_dev_API yarp::dev::IJoypadEventDriven : yarp::os::RateThread,
 
 
     IJoypadEventDriven();
-
-    IJoypadEventDriven(int rate);
+#ifndef YARP_NO_DEPRECATED // Since YARP 3.0.0
+    explicit IJoypadEventDriven(YARP_DEPRECATED_MSG("Use IJoypadEventDriven(double)") int rate);
```

Comment:
Here the explanation of the strange position of `YARP_DEPRECATED_MSG`:

- https://stackoverflow.com/questions/46018977/what-is-the-correct-syntax-for-deprecating-a-single-constructor-in-visual-studio
- https://stackoverflow.com/questions/38378693/did-visual-studio-2015-update-3-break-constructor-attributes